src/axom/core/utilities/System.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/core/utilities/System.hpp"
+
+#ifdef WIN32
+  #include <windows.h>
+  #include <tchar.h>
+#else
+  #include <unistd.h>
+  #include <limits.h>
+  #include <pwd.h>
+#endif
+
+#include <iostream>
+
+namespace axom
+{
+namespace utilities
+{
+#define INFO_BUFFER_SIZE 32767
+
+std::string getHostName()
+{
+  std::string hostName = "";
+#ifdef WIN32
+  TCHAR infoBuf[INFO_BUFFER_SIZE];
+  DWORD bufCharCount = INFO_BUFFER_SIZE;
+  bufCharCount = INFO_BUFFER_SIZE;
```

Comment:
what does line 31 do that didn't get done by line 30?
edit: Also line 52 vs line 51.  If there is a reason for the reassignment, would you please put a brief comment saying this is indeed necessary (and perhaps why it is)?

Reply:
I copy/pasted this chunk from microsoft: https://docs.microsoft.com/en-us/windows/win32/sysinfo/getting-system-information

Reply:
though that seems maybe its there for clarity? ill remove it.

Reply:
microsoft!?!  surely, that will cause problems for users...